portfolio/src/main/webapp/script.js

```diff
@@ -11,18 +11,74 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+let init = false, projects, projectNameEl, projectTextEl, projectLinkEl, projectImgEl;
+
+/** Initializes all the variables needed for addRandomProject */
+function populate() {
+  projects = [
+      { name:"Tak", 
+        text:"This is terminal version of the KingKiller Chronicle's two player game with the same name!", 
+        link:"https://github.com/connielei-stuy/apcs1-final-tak", 
+        image:"/images/tak.png"
+      }, 
+      { name:"TriWizard Maze", 
+        text:"This is virtual reality experience of the TriWizard Maze!", 
+        link:"https://github.com/connielei-kwk/kwk17-FINAL-PROJ-maze", 
+        image:"/images/tri.png"
+      },
+      { name:"Graphics Engine", 
+        text:"This is simple graphics engine that can create basic shapes.", 
+        link:"https://github.com/connielei-stuy/graphics-anim", 
+        image:"/images/graphics.gif"
+      } 
+  ];
+  projectNameEl = document.getElementById('project-name');
+  projectTextEl = document.getElementById('project-text');
+  projectLinkEl = document.getElementById('project-link');
+  projectLinkEl.innerText = "Click here for the repo!";
+  projectImgEl = document.getElementById('project-img');
+  init = true;
+}
+
+/** Updates the project name element with the given name. */
+function updateProjectName(name) {
+  projectNameEl.innerText = name;
+}
+
+/** Updates the project text element with the given text. */
+function updateProjectText(text) {
+  projectTextEl.innerText = text;
+}
+
+/** Upadtes the project link element with the given link. */
+function updateProjectLink(link) {
+  projectLinkEl.href = link;
+}
+
+/** Updates the image element with the given image. */
+function updateProjectImage(image) {
+  projectImgEl.src = image;
+}
+
+/** Returns a random project. */
+function getRandomProject() {
+  return projects[Math.floor(Math.random() * projects.length)];
+}
 
 /**
- * Adds a random greeting to the page.
+ * Initializes all necessary variables if needed, then
```

Comment:
nit: The convention is to wrap lines at 80 columns; most editors have a way of setting up a ruler that shows up at that location to make it easier for you to know where to wrap!